x-pack/solutions/observability/plugins/apm/public/components/shared/trace_waterfall/badges/constants.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticAgentName } from '@kbn/elastic-agent-utils/src/agent_names';
+import {
+  OPEN_TELEMETRY_AGENT_NAMES,
+  EDOT_AGENT_NAMES,
+} from '@kbn/elastic-agent-utils/src/agent_names';
+
+type OtelLanguage = 'webjs' | 'swift' | 'android' | 'cpp' | 'erlang' | 'rust';
+type SyncBadgeAgentName = ElasticAgentName | OtelLanguage;
+
+/**
+ * Set of known OTEL and EDOT agents for O(1) lookup validation.
+ */
+const knownOtelAgents: Set<string> = new Set([...OPEN_TELEMETRY_AGENT_NAMES, ...EDOT_AGENT_NAMES]);
+
+/**
+ * Maps agent names and language names to their expected sync behavior.
+ * Includes Elastic agent names and OTEL language names.
+ *
+ * true: Language natively operates synchronously (e.g., Node.js, browser JS)
+ * false: Language natively operates asynchronously (e.g., Python, PHP, Go)
+ */
+export const agentsSyncMap = {
```

Comment:
Couldn't this be simplified to be a Set as well?

EDIT: Thinking about it, just make this an actual Map, if you need to test for presence of the key, and then resolve a value.